sklearn/metrics/tests/test_pairwise.py

```diff
@@ -370,6 +372,49 @@ def test_pairwise_distances_argmin_min():
     np.testing.assert_almost_equal(dist_orig_val, dist_chunked_val, decimal=7)
 
 
+def check_invalid_block_size_generator(generator):
+    for i in generator:
+        return i
+
+
+def test_pairwise_distances_blockwise_invalid_block_size():
+    rng = np.random.RandomState(0)
+    X = rng.random_sample((400, 4))
+    y = rng.random_sample((200, 4))
+    gen = pairwise_distances_blockwise(X, y, block_size=0, metric='euclidean')
+    assert_raise_message(ValueError, 'block_size should be at least n_samples '
```

Comment:
I'd rather this error be raised upon calling `pairwise_distances_blockwise`. That means you need to not use a generator function immediately, but use a secondary function call.